examples/swarm/test_db.py

```diff
@@ -19,41 +19,103 @@
+USER_CONFIG = "nupic-site.xml"
+
+def getFileUsed():
+  """
+    Determine which NuPIC configuration file is being used and returns the
+    name of the configuration file it is using. Either DEFAULT_CONFIG or
+    USER_CONFIG.
+  """
+
+  # output will be {} if the file passed into Configuration._readConfigFile
+  # can not be found in the standard paths returned by
+  # Configuration._getConfigPaths.
+  output = Configuration._readConfigFile(USER_CONFIG) #pylint: disable=protected-access
+  if output != {}:
+    return USER_CONFIG
+  return DEFAULT_CONFIG
+
+def testDbConnection(host, port, user, passwd):
+  """
+    Determine if the specified host, port, user, passwd is able to connect
+    to a running mysql database, create test database, create a test table,
+    insert something into the table, delete the table, and delete the database.
+    returns true if this is successful, false if there is an error in this
+    process.
+  """
+  try:
+    conn = pymysql.connect(host=host, port=port, user=user, passwd=passwd)
+    cursor = conn.cursor()
+    cursor.execute("CREATE DATABASE IF NOT EXISTS nupic_db_test")
+    conn.select_db("nupic_db_test")
+    cursor.execute("CREATE TABLE db_test \
+                              (teststring VARCHAR(255),\
+                               someint INT)")
+    cursor.execute("INSERT INTO db_test VALUES ('testing123', 123)")
+    cursor.execute("DROP TABLE IF EXISTS db_test")
+    cursor.execute("DROP DATABASE IF EXISTS nupic_db_test")
+    return True
+
+  except pymysql.err.OperationalError:
+    return False
+
+
+def dbValidator():
+  """
+    Let the user know what NuPIC config file is being used
+    and whether or not they have mysql set up correctly for
+    swarming.
+  """
+  fileused = getFileUsed()
+
+  # Get the values we need from NuPIC's configuration
+  host = Configuration.get("nupic.cluster.database.host")
+  port = int(Configuration.get("nupic.cluster.database.port"))
+  user = Configuration.get("nupic.cluster.database.user")
+  passwd = Configuration.get("nupic.cluster.database.passwd")
+
+
+  print "This script will validate that your MySQL is setup correctly for "
+  print "NuPIC. MySQL is required for NuPIC swarming. The settings are"
+  print "defined in a configuration file found in  "
+  print "$NUPIC/src/nupic/support/nupic-default.xml Out of the box those "
+  print "settings contain MySQL's default access credentials."
+  print
+  print "The nupic-default.xml can be duplicated to define user specific "
+  print "changes calling the copied file "
+  print "$NUPIC/src/nupic/support/nupic-site.xml Refer to the "
+  print "nupic-default.xml for additional instructions."
+  print
+  print "Defaults: localhost, 3306, root, no password"
+  print
+  print "Retrieved the following NuPIC configuration using: ", fileused
+  print "    host   :    ", host
+  print "    port   :    ", port
+  print "    user   :    ", user
+  print "    passwd :    ", passwd
```

Comment:
Damn, I should have noticed this before, but some people will not appreciate their plaintext password being printed to the console. I suggest you remove this line or replace it with `XXXXXXXX`.